examples/secret_channel.py

```diff
@@ -0,0 +1,106 @@
+
+@create.command()
+@can_create()
+async def voice(ctx, *, new_channel: str):
+    """Subcommand to create an open voice channel."""
+
+    # note: ctx.guild.categories will return a list of possible categories that the server has
+    # you can then select where the channel will be created in the category list
+    channel = await ctx.guild.create_voice_channel(new_channel, category=ctx.guild.categories[1])
+    await ctx.send(f"Voice channel {new_channel} was created!") # will send into channel that command was invoked from
+
+@create.command()
+@can_create()
+async def priv(ctx, *, new_channel: str):
+    """Subcommand to create a private text channel."""
+
+    # use discord.utils.get() for retrieving and storing a role into variables
+    # guild.roles is an iterable, and name is an attribute to search for
+    admin = get(ctx.guild.roles, name="YOUR_ADMIN_ROLE")
+    mods = get(ctx.guild.roles, name="YOUR_MOD_ROLE")
+
+    # using a dictionary, permissions can be chosen for the new channel
+    # guild.default_role is @everyone, guild.me is the bot itself
+    # using admin and mods allows to include them into the new channel
+    overwrites = {
+        ctx.guild.default_role: discord.PermissionOverwrite(read_messages=False),
+        ctx.guild.me: discord.PermissionOverwrite(read_messages=True),
+        admin: discord.PermissionOverwrite(read_messages=True),
+        mods: discord.PermissionOverwrite(read_messages=True)
+    }
+    
+    # the overwrites dict is assigned to the overwrites parameter 
+    channel = await ctx.guild.create_text_channel(new_channel, overwrites=overwrites)
+    await channel.send(f"Private text channel {new_channel} was created!")
+
+@create.command()
+@can_create()
+async def priv_voice(ctx, *, new_channel: str):
+    """Subcommand to create a private voice channel."""
+    
+    admin = get(ctx.guild.roles, name="YOUR_ADMIN_ROLE")
+    mods = get(ctx.guild.roles, name="YOUR_MOD_ROLE")
+
+    overwrites = {
+        ctx.guild.default_role: discord.PermissionOverwrite(read_messages=False),
+        ctx.guild.me: discord.PermissionOverwrite(read_messages=True),
+        admin: discord.PermissionOverwrite(read_messages=True),
+        mods: discord.PermissionOverwrite(read_messages=True)
+    }
+
+    # using create_voice_channel to create a new voice channel
+    # the parameters are used in the same manner as the create_text_channel command
+    channel = await ctx.guild.create_voice_channel(new_channel, overwrites=overwrites)
+    await ctx.send(f"Private voice channel {new_channel} was created!")
+
+@bot.command()
+@can_create()
+async def delete(ctx, *, channel_name: str):
+    """Command to delete a given channel."""
+
+    # using discord.utils.get() and bot.get_all_channels(), you can specify an attribute
+    # to search through an iterable, in this case all the channels on a guild
+    channel = get(bot.get_all_channels(), name=channel_name)
```

Comment:
This seems exceedingly dangerous for an example.

Reply:
Additionally, your comment suggests `in this case all the channels on a guild` but this is not the case with your code, you are using `.get_all_channels` which returns all the channels on *all* of the guilds the bot can see